_posts/2023-01-22-auto-restarting-connectors.md

```diff
@@ -0,0 +1,317 @@
+Thanks to this back-off mechanism, even if the network outage takes over 10 minutes, the auto-restart will help your connectors to recover from it.
+The last 7th restart will happen approximately only after 112 minutes after the initial failure.
+But if the issue is not resolved even after the 7th restart, the operator will stop restarting and it is up to you to solve it manually.
+
+You can track the restarting progress in the `.status` section of the `KafkaConnector` resource:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnector
+metadata:
+  name: my-connector
+spec:
+  # ...
+status:
+  # ...
+  autoRestart:
+    count: 1
+    lastRestartTimestamp: "2023-01-22T21:38:24.402461310Z"
+```
+
+If the connector recovers and keeps running, the counter of the restarts is reset as well.
+And when the next issue happens - possibly days or weeks later - it will start again from 0.
+
+### Example
+
+Let's have a look at how it works in an example.
+First, you have to install Strimzi 0.33.0 or newer and deploy a Kafka cluster.
+In my case, I used to simplest possible cluster based on our examples:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    version: 3.3.2
+    replicas: 3
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
+      inter.broker.protocol.version: "3.3"
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+  zookeeper:
+    replicas: 3
+    storage:
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: false
+  entityOperator:
+    topicOperator: {}
+    userOperator: {}
+```
+
+Next, we create a topic that we will use:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 3
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+```
+
+Then, we have to deploy the Kafka Connect cluster.
+It enables the connector operator with the `strimzi.io/use-connector-resources` annotation.
+And it adds a custom [Echo Sink connector](https://github.com/scholzj/echo-sink) to the deployment.
+The Echo Sink connector is my _test_ connector which gets the messages from Kafka Connect, but instead of sending them to some external system, it simply logs them to the standard output.
+It has also a special option `fail.task.after.records` which makes the connector tasks fail after receiving a pre-configured amount of messages.
+And we will use this feature to demonstrate the auto-restart feature.
```

Comment:
Maybe mention this after the `KafkaConnect` resource, just before the `KafkaConnector` resource?